pandas/core/frame.py

```diff
@@ -5777,13 +5777,14 @@ def groupby(
         group_keys: bool = True,
         squeeze: bool = False,
         observed: bool = False,
-    ) -> "groupby_generic.DataFrameGroupBy":
+    ) -> "DataFrameGroupBy":
+        from pandas.core.groupby.generic import DataFrameGroupBy
```

Comment:
Can this no longer be done at the top of the module rather than in the function?

Reply:
Its not a good idea to import the class at the top because of circular import problems.

There was a discussion in #30314 if the import could be less ugly (generic_groupby is not a module name). This is an attempt to improve on that and makes the import for groupbys and resample more similar to each other.

Reply:
Gotcha. Generally not a fan of this but I guess somewhat orthogonal